### PR TITLE
tests: add functions to make an abstraction for the snaps

### PR DESCRIPTION
**Pull request by sergiocazzolato**
This is the first part of the change to avoid deciding which snap to use
based on the core system which is being used.

### CODE REVIEW

File: tests/lib/systems.sh [hunk 1]
```diff
@@ -27,3 +34,28 @@ is_ubuntu_14_system(){
     fi
     return 1
 }
+
+get_snap_for_system(){
+    snap=$1
+    system=${2:-$SPREAD_SYSTEM}
+
+    if [[ "$system" == ubuntu-core-18-* ]]; then
```
Comment (zyga) [inline tests/lib/systems.sh]: Perhaps use a switch statement here, it's a natural fit for this kind of code.

Comment (sergiocazzolato) [reply tests/lib/systems.sh]: Totally agree, thanks for the review

File: tests/lib/systems.sh [hunk 2]
```diff
@@ -14,16 +14,55 @@ is_core18_system(){
...
+
 is_ubuntu_14_system(){
     if [[ "$SPREAD_SYSTEM" == ubuntu-14.04-* ]]; then
         return 0
     fi
     return 1
 }
+
+get_snap_for_system(){
+    snap=$1
```
Comment (zyga) [inline tests/lib/systems.sh]: If this is bash then we should be more careful with variables and use `local`. Alternatively, if you can, avoid the variables altogether, when it is natural to do so.

File: tests/lib/systems.sh [hunk 3]
```diff
@@ -14,16 +14,55 @@ is_core18_system(){
...
+
+    case "$system" in
+        ubuntu-core-18-*)
+            echo "core18"
+            ;;
+        ubuntu-core-20-*)
+            echo "core20"
+            ;;
+        *)
+            echo "core"
```
Comment (zyga) [inline tests/lib/systems.sh]:
```suggestion
            echo "core"
            ;;
```

File: tests/lib/systems.sh [hunk 4]
```diff
@@ -14,16 +14,55 @@ is_core18_system(){
...
+
+    case "$system" in
+        ubuntu-core-18-*)
+            echo "${snap}-core18"
+            ;;
+        ubuntu-core-20-*)
+            echo "${snap}-core20"
+            ;;
+        *)
+            echo "$snap"
```
Comment (zyga) [inline tests/lib/systems.sh]:
```suggestion
            echo "$snap"
            ;;
```

File: tests/lib/systems.sh [hunk 5]
```diff
@@ -14,16 +14,55 @@ is_core18_system(){
     return 1
 }
 
+is_core20_system(){
+    if [[ "$SPREAD_SYSTEM" == ubuntu-core-20-* ]]; then
```
Comment (zyga) [inline tests/lib/systems.sh]: I wish there was a way to make this a one-liner without upsetting `set -e`.

File: tests/lib/systems.sh (hunk 4 above)
Comment (sergiocazzolato) [reply tests/lib/systems.sh]: done

File: tests/lib/systems.sh (hunk 3 above)
Comment (sergiocazzolato) [reply tests/lib/systems.sh]: done

File: tests/lib/systems.sh (hunk 2 above)
Comment (sergiocazzolato) [reply tests/lib/systems.sh]: Sure, thanks for reviewing this.

**Comment by sergiocazzolato** [comment]
@zyga could you make another round please?

File: tests/lib/systems.sh [hunk 6]
```diff
@@ -14,16 +14,57 @@ is_core18_system(){
...
 is_ubuntu_14_system(){
     if [[ "$SPREAD_SYSTEM" == ubuntu-14.04-* ]]; then
         return 0
     fi
     return 1
 }
+
+get_snap_for_system(){
+    local snap=$1
+    local system=${2:-$SPREAD_SYSTEM}
```
Comment (zyga) [inline tests/lib/systems.sh]: I'm kind of -1 on the second argument unless there's a really good reason to have it.

Comment (sergiocazzolato) [reply tests/lib/systems.sh]: done

Comment (sergiocazzolato) [reply tests/lib/systems.sh]: done

**Pull request merged**